File: src/main/java/com/learnershi/rclasssocket/example/WSConfig.java
package com.learnershi.rclasssocket.example;

import org.springframework.context.annotation.Configuration;
import org.springframework.messaging.simp.config.MessageBrokerRegistry;
import org.springframework.web.socket.config.annotation.EnableWebSocketMessageBroker;
import org.springframework.web.socket.config.annotation.StompEndpointRegistry;
import org.springframework.web.socket.config.annotation.WebSocketMessageBrokerConfigurer;

@Configuration
@EnableWebSocketMessageBroker
public class WSConfig implements WebSocketMessageBrokerConfigurer {

    @Override
    public void configureMessageBroker(MessageBrokerRegistry config) {
        config.enableSimpleBroker("/");
        config.setApplicationDestinationPrefixes("/app");
    }

    @Override
    public void registerStompEndpoints(StompEndpointRegistry registry) {
        registry.addEndpoint("/gs-guide-websocket")
                .setAllowedOrigins("http://127.0.0.1:5173")
                .withSockJS();
    }
}

File: src/main/java/com/learnershi/rclasssocket/config/WSConfig.kt
package com.learnershi.rclasssocket.config

import com.learnershi.rclasssocket.log.Log
import org.springframework.context.annotation.Configuration


@Configuration
//@EnableWebSocketMessageBroker
class WSConfig {
    val log = Log.of(this.javaClass)
//    @Bean
//    fun rsocketMessageHandler() = RSocketMessageHandler().apply {
//        routeMatcher = PathPatternRouteMatcher()
//    }
}